bluehoover/clickhouse.py

```diff
@@ -31,9 +31,16 @@ def __init__(
         self.post_queue = asyncio.Queue()
         self.batch_size = batch_size
         self.timeout = timeout
-        self.cursor = self.get_checkpoint()
+        self.cursor = None
         self.websocket_task = None
         self.running = True
+        self.clickhouse = ClickHouseManager()
+        if "KeepermapUpdating" == CHECKPOINT_METHOD:
+            self.checkpoint = KeepermapUpdatingCheckpoint()
+        elif "KeepermapDeleting" == CHECKPOINT_METHOD:
+            self.checkpoint = KeepermapDeletingCheckpoint()
```

Comment:
You might have good reason to do it this way, but I'd normally go:

```suggestion
        if CHECKPOINT_METHOD == "KeepermapUpdating" :
            self.checkpoint = KeepermapUpdatingCheckpoint()
        elif CHECKPOINT_METHOD == "KeepermapDeleting":
            self.checkpoint = KeepermapDeletingCheckpoint()
```

Reply:
It's an old habit called Yoda conditions. The idea is to always put the literal on the left side of the comparison operator. In the case of a typo (`=` instead of `==`), assigning to a literal is a syntax error whereas assigning to a variable you meant to compare can create some really pernicious bugs.